### PR TITLE
IDENTITY-6032: Pass HTTP Request Headers to Grant Handlers

### PR DESCRIPTION
**Pull request by mefarazath**
https://wso2.org/jira/browse/IDENTITY-6032
(This involves API additions)

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/HttpRequestHeader.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package org.wso2.carbon.identity.oauth2.model;
+
+import java.io.Serializable;
+
+/**
+ * This class is used to store http request header information
+ */
+public class HttpRequestHeader implements Serializable {
```
Comment (Pushpalanka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/HttpRequestHeader.java]: Serializable class should have a static final serialVersionUID field of type long. In IDEA there is a plugin to auto generate this.

**Comment by mefarazath** [comment]
Added **serialVersionUID** to the HttpRequestHeader class.

**Comment by mefarazath** [comment]
@isharak @Pushpalanka @kasunbg 

Today I during a discussion related to implementing token binding I realized that we need to pass headers to underlying grant handlers and other extensions points. 

According to what I understood in TLS binding, we make use of information sent in headers to validate refresh token etc. So can we consider merging this PR to facilitate passing header information to downstream extensions from the TokenEndpoint?

**Comment by kasunbg** [comment]
Hi @mefarazath,

Yes, we are implementing TLS Token Binding with a [TTRP](https://tools.ietf.org/html/draft-ietf-tokbind-ttrp-01) in the first phase. In this, a HTTP header named "Sec-Provided-Token-Binding-Id" and "Sec-Referred-Token-Binding-Id" will be sent to the back-end Identity Server. 

These headers need to be validated by the grant types. So, the grant type handlers must receive these HTTP headers.

Thanks.

**Comment by kasunbg** [comment]
This is an API addition, and the existing signatures are not changed. So, we should do a minor version bump when merging this PR.

**Pull request merged**